### PR TITLE
Main view from YTS

### PR DESCRIPTION
**Pull request by apascual**

### CODE REVIEW

File: PopcornTime/Source/Player/Views/MainView.swift [hunk 1]
```diff
@@ -0,0 +1,111 @@
+//
+//  MainView.swift
+//  PopcornTime
+//
+//  Created by Abel Pascual on 28/01/16.
+//  Copyright © 2016 com.PPinera. All rights reserved.
+//
```
Comment (pepibumur) [inline PopcornTime/Source/Player/Views/MainView.swift]: @apascual avoid headers in the project

File: PopcornTime/Source/Player/Views/MainView.swift [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+import UIKit
+import Alamofire
+
```
Comment (pepibumur) [inline PopcornTime/Source/Player/Views/MainView.swift]: Empty line

File: PopcornTime/Source/Player/Views/MainView.swift [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+import AVKit
+import AVFoundation
+import PopcornTorrent
+
+
+class MainView: UIViewController, UICollectionViewDelegate, UICollectionViewDataSource {
+    
+    @IBOutlet weak var collectionView: UICollectionView!
+
+    private var _movies: NSArray?
```
Comment (pepibumur) [inline PopcornTime/Source/Player/Views/MainView.swift]: You don't need duplicated variables here:

```swift
private var movies: NSArray {
  set {
    self.movies = newValue
    self.collectionView.reloadData()
  }
}
```

File: PopcornTime/Source/Player/Views/MainView.swift [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        }
+        set {
+            self._movies = newValue
+            self.collectionView.reloadData()
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
```
Comment (pepibumur) [inline PopcornTime/Source/Player/Views/MainView.swift]: empty line

File: PopcornTime/Source/Player/Views/MainView.swift [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        loadMovies()
+    }
+    
+    func loadMovies() {
+        Alamofire.request(.GET, "https://yts.ag/api/v2/list_movies.json?sort_by=download_count", parameters: nil, encoding:.JSON).responseJSON
```
Comment (pepibumur) [inline PopcornTime/Source/Player/Views/MainView.swift]: There's a framework that already implements that @apascual : 

```swift
import PopcornKit
```

This is the factory of commands https://github.com/PopcornTimeTV/PopcornTimeTV/blob/master/PopcornKit/Source/Factories/YTSCommandFactory.swift

Once you have the command you can schedule it in a background queue, look at the example here: https://github.com/pepibumur/HTTPCommander

File: PopcornTime/Source/Player/Views/MainView.swift [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+                print("Request failed with error: \(error)")
+                }
+        }
+    }
+    
+    
+    /*
+    // MARK: - Navigation
+    
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
```
Comment (pepibumur) [inline PopcornTime/Source/Player/Views/MainView.swift]: Avoid comments. Code should be explicit enough.

File: PopcornTime/Source/Player/Views/MainView.swift [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
+    // Get the new view controller using [segue destinationViewController].
+    // Pass the selected object to the new view controller.
+    }
+    */
+    
+    // MARK: UICollectionViewDataSource
+    
+    func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return self.movies?.count ?? 0
```
Comment (pepibumur) [inline PopcornTime/Source/Player/Views/MainView.swift]: I would force to be non-optional value.

File: PopcornTime/Source/Player/Views/MainView.swift [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+    }
+    
+    func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCellWithReuseIdentifier("movieCell", forIndexPath: indexPath)
+        
+        let movieDic = self.movies![indexPath.row] as! NSDictionary
```
Comment (pepibumur) [inline PopcornTime/Source/Player/Views/MainView.swift]: When you use the command I proposed you won't have to deal with dictionaries anymore but directly accessing models and its attributes. Much better!

File: PopcornTime/Source/Player/Views/MainView.swift [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+            (cell as! MovieCollectionViewCell).posterImageView.image = image
+        }
+        
+        return cell
+    }
+    
+    func collectionView(collectionView: UICollectionView,
+        layout collectionViewLayout: UICollectionViewLayout,
+        sizeForItemAtIndexPath indexPath: NSIndexPath) -> CGSize {
+            return CGSizeMake(213, 380)
```
Comment (pepibumur) [inline PopcornTime/Source/Player/Views/MainView.swift]: Extrac these values into constants.

File: PopcornTime/Source/Player/Views/MainView.swift [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+    }
+    
+    func collectionView(collectionView: UICollectionView, didSelectItemAtIndexPath indexPath: NSIndexPath) {
+        let movieDic = self.movies![indexPath.row] as! NSDictionary
+        let torrentURL = movieDic["torrents"]?[0]["url"] as! String
+
+        let playerView = PlayerView()
+        playerView.torrentUrl = torrentURL
+        self .presentViewController(playerView, animated: true) { () -> Void in
+            //
```
Comment (pepibumur) [inline PopcornTime/Source/Player/Views/MainView.swift]: ??

File: PopcornTime/Source/Player/Views/MovieCollectionViewCell.swift [hunk 11]
```diff
@@ -0,0 +1,16 @@
+//
+//  MovieCollectionViewCell.swift
+//  PopcornTime
+//
+//  Created by Abel Pascual on 28/01/16.
```
Comment (pepibumur) [inline PopcornTime/Source/Player/Views/MovieCollectionViewCell.swift]: Remove headers

File: PopcornTime/Source/Player/Views/PlayerView.swift [hunk 12]
```diff
@@ -3,12 +3,14 @@ import AVKit
 import AVFoundation
 import PopcornTorrent
```
Comment (pepibumur) [inline PopcornTime/Source/Player/Views/PlayerView.swift]: Empty space

File: PopcornTime/Source/Player/Views/PlayerView.swift [hunk 13]
```diff
@@ -3,12 +3,14 @@ import AVKit
...
 
+import Alamofire
+
 internal class PlayerView: AVPlayerViewController {
 
     // MARK: - Attributes
     
     internal var presenter: PlayerPresenter?
-
```
Comment (pepibumur) [inline PopcornTime/Source/Player/Views/PlayerView.swift]: I would first do the home view and them think about the player view. Try to have PR with an unique purpose. Let's leave this for a different PR

**Comment by pepibumur** [comment]
Left some comments @apascual , good start. You should be able to push directly this branch into the repository since I added you as a contributor. Once you review these changes I'll help you with some code style to make it more nicer but good starting point. It's very helpful :package:

**Comment by OstlerDev** [comment]
@pepibumur I think that for sake of clarity there should be many comments throughout the code, explaining what it does a bit. I am not meaning over commenting, but for any complex or vague items (especially places with magic numbers) comments should be included.
